src/firefly/js/visualize/ui/extraction/ExtractionTable.jsx

```diff
@@ -84,46 +91,186 @@ export function keepZAxisExtraction(pt, pv, plot, filename, refHDUNum, extractio
     save ? doDispatchTableSaving(dataTableReq, doOverlay) : doDispatchTable(dataTableReq, doOverlay);
     idCnt++;
     titleCnt++;
-    return tbl_id;
+    return Promise.resolve([tbl_id]);
 }
 
-export function keepLineExtraction(pt, pt2, pv, plot, filename, refHDUNum, plane, extractionSizeX, extractionSizeY, combineOp, save = false, doOverlay = true) {
+export function keepDataExtraction({pv, baseImPtAry, save, doOverlay, axis='both', pointSize, combineOp, isLine=false}) {
+    const plot= primePlot(pv);
+    const filename= plot?.plotState?.getWorkingFitsFileStr();
     if (!pv || !plot || !filename) {
         showInfoPopup('Plot no longer exist. Cannot extract.');
         return;
     }
+    const {pvAry,filteredPvAry,canDoMultiImage}= getExtractableImageList(pv);
+    const extractionSizeX= axis==='x' ? 1 : pointSize;
+    const extractionSizeY= axis==='y' ? 1: pointSize;
+    const params= {baseImPtAry, pv, extractionSizeX, extractionSizeY, combineOp, save, doOverlay, isLine};
+    if (!canDoMultiImage) {
+        return Promise.resolve([makeDataExtractionTable(params)]);
+    }
+
+    if (filteredPvAry.length < pvAry.length) {
+        let msg= 'Cannot extract from all images';
+        if (pvAry.some( (testPv) => !hasWCSProjection(testPv))) {
+            msg+= ' Some image do not have a WCS Projection.';
```

Comment:
minor: `'Some images do not have a WCS Projection.'` 
`images` should be plural is the suggested change